Add `jsdoc` bundle package

tomalec opened the pull request:
### Changes proposed in this Pull Request:

Addresses some of the concerns from https://github.com/woocommerce/grow/pull/6#pullrequestreview-984688504

Bundles multiple commonly used `jsdoc` plugins, so the extensions repo would need only this one dependency.
Exports a binary, that wraps commonly used `jsdoc` CLI arguments, for easier use.




### Screenshots:

<!--- Optional --->


### Detailed test instructions:
<!-- Add detailed instructions for how to test that this PR fixes the issue and confirm that it doesn't break any other features :) -->

See example usage at https://github.com/woocommerce/google-listings-and-ads/compare/develop...tweak/bundled-jsdoc
And the installation & usage instructions at https://github.com/woocommerce/grow/blob/add/jsdoc-bundle/packages/js/jsdoc/README.md 

1. `npm i --save-dev https://gitpkg.now.sh/woocommerce/grow/packages/js/jsdoc?add/jsdoc-bundle`
2. Create a `TRACKING.md` file
   ```md 
   # Usage Tracking
   Some nice general description.
   <woocommerce-grow-tracking-jsdoc></woocommerce-grow-tracking-jsdoc>
   ```
4. Create a `.jsdocrc.json`
   ```json
   {
      "plugins": [
         "woocommerce-grow-jsdoc/tilde-alias",
         "woocommerce-grow-tracking-jsdoc/fires-description",
         "jsdoc-plugin-typescript",
         "jsdoc-advanced-types-plugin",
         "jsdoc-plugin-intersection"
      ],
      "typescript": {
         "moduleRoot": "js/src"
      }
   }
   ```
6. `npx woocommerce-grow-jsdoc`


### Additional details:

<!--
Optional.
Enter a summary of all changes in this Pull Request, which will be added to the changelog if accepted.
Each line should start with `(Fix|Add|Tweak|Update) - `, for example:
> Fix - I took care of something that wasn't working.
> Add - I added something new that's pretty cool.
> Tweak - I made a small change.
> Update - I made big changes to something that wasn't broken.

Leave the "Changelog entry" header in place completely empty, without any summary if no changelog entry is needed.
If you remove the "Changelog entry" header, the title of Pull Request will be used as the changelog entry.
-->
### Changelog entry

> Add - `-jsdoc` bundle package

tomalec committed:
Initialize `woocommerce-grow-jsdoc` bundle

tomalec committed:
Bin in jsdoc bundle

(cherry picked from commit cb88933b5f26747d9f3ca4d94c8a625959e695e6)
Co-authored-by: Miguel Perez Pellicer <5908855+puntope@users.noreply.github.com>

tomalec committed:
Remove shared `utils.js`,

as npm fails to resolve shelljs dependency.

tomalec committed:
Install `-tracking-jsdoc` as `-jsdoc`s dependency

tomalec committed:
Describe bundled `-jsdoc` plugins

tomalec committed:
Move `tilde-alias` jsdoc plugin to `-jsdoc`,

from `-tracking-jsdoc`.
It has nothing to do with Tracking, it's just our code convention.

tomalec committed:
Remove the "one plugin" from the `-jsdoc` README,

as it is not implemented.

tomalec committed:
Remove the unrelated installation instruction

from the `-jsdoc` readme.
The plugin should be already installed given the budnle is.

puntope commented on packages/js/storybook/bin/storybook.js:
```diff
@@ -3,11 +3,10 @@
- * Internal dependencies
- */
-import { shell } from '../utils.js'
+process.env.PATH += ( path.delimiter + path.join( process.cwd(), 'node_modules', '.bin') );
```
This line is repeated over all the bin files, so I guess could be reuse it in `utils.js` kinda file.

tomalec replied:
In all 2 ;) We cannot.
as npm fails to resolve shelljs dependency in a parent folder without its own package.json
https://github.com/woocommerce/grow/pull/10/commits/99141963a638199522a213fdda065f05b7847a9a

Plus it's single line
```js
/**
 * Internal dependencies
 */
import { shell } from '../../utils.js'
```
vs.
```js
import shell from 'shelljs';
import path from 'path';

process.env.PATH += ( path.delimiter + path.join( process.cwd(), 'node_modules', '.bin') );
```

So not a big difference in LOC, and a bit more explicit, that it's just the original `shelljs`

puntope commented:
⚠️ ❓  

When I ran the installation I got this error:

```
➜  npm i --save-dev "https://gitpkg.now.sh/woocommerce/grow/packages/js/jsdoc?add/jsdoc-bundle"
npm ERR! code ENOLOCAL
npm ERR! Could not install from "node_modules/tracking-jsdoc" as it does not contain a package.json file.

npm ERR! A complete log of this run can be found in:
npm ERR!     /Users/miguelperezpellicer/.npm/_logs/2022-06-07T12_58_34_945Z-debug.log
```

tomalec committed:
Use gitpkg reference to `woocommerce-grow-tracking-jsdoc`

instead of relative FS path, to make the `jsdoc` package installable elsewhere.

Related to https://github.com/woocommerce/grow/issues/5

Address https://github.com/woocommerce/grow/pull/10#issuecomment-1148644082

tomalec committed:
Use `git+https` instead of `github:` protocol for `-jsdoc` dependency.

tomalec commented:
> When I ran the installation I got this error:

Thanks, it was a leftover from my local testing with a relative filesystem reference. Fixed in https://github.com/woocommerce/grow/pull/10/commits/8b88bfdee0d03cf5bd1539cb2856e3dc7bcfa2e2

tomalec commented:
@puntope would you mind another look?

puntope reviewed (approved):
✅  LGTM

I was able to install the plugin and generate the tracking docs 

Thanks ❤️
